ext/random/tests/03_randomizer/basic.phpt

```diff
@@ -27,6 +27,18 @@ $engines[] = new UserEngine();
 foreach ($engines as $engine) {
     $randomizer = new Random\Randomizer($engine);
 
+    // nextInt
+    for ($i = 0; $i < 1000; $i++) {
+        try {
+            $randomizer->nextInt();
+        } catch (\RuntimeException $e) {
+            if ($e->getMessage() !== 'Generated value exceeds size of int') {
+                die($engine::class . ': nextInt: failure: {$e->getMesasge()}');
+                throw $e;
```

Comment:
- before this `if`, others similar tests first do a
  ```php
  if (\PHP_INT_SIZE >= 8) {
      throw $e;
  }
  ```
- typo in the `die`: <code>$e->getMes**as**ge()</code> should be <code>$e->getMes**sa**ge()</code>
- bug in the `die`: single-quotes should be double-quotes, or (maybe better) replace string interpolation with concatenation
- this `throw` is unreachable (after a `die`)

Reply:
fixed: d78c647ca419312e7fc8dbea6c16a06a34bb9f5b

Reply:
OK for the fixes (of my last 3 points),
but I don't understand your last comment (about my first point) 😕 I saw https://github.com/php/php-src/blob/b358834c7221bf2ca5a2325da7f7055ca7d2cdb3/ext/random/tests/03_randomizer/compatibility_user.phpt#L45-L52 (2 occurrences in that file) so was wondering if here you should do something like
```diff
         } catch (\RuntimeException $e) {
+            if (\PHP_INT_SIZE >= 8) {
+                die($engine::class . ": nextInt: failure: {$e->getMessage()}");
+            }
             if ($e->getMessage() !== 'Generated value exceeds size of int') {
                 die($engine::class . ": nextInt: failure: {$e->getMessage()}");
             }
        }
```
? or maybe merge the two `if`s into one `if ($e->getMessage() !== 'Generated value exceeds size of int' || \PHP_INT_SIZE >= 8)` (in both files)? (just questions BTW, not blocking this PR)

Reply:
Sorry. I was confused again.
This is completely unnecessary and should be removed. If it is a good thing to modify at the same time as this PR, I will do so now. 😭

Reply:
> Sorry. I was confused again.

No big deal 😄

> This is completely unnecessary and should be removed.

Well, I thought that it was at least testing that nextInt() doesn't throw a "Generated value exceeds size of int" exception on 64-bit systems, but you know better 😉

Reply:
fixed: 7c41dd9c9a96932b7acbd811bfb7d1cca7c142cc

Reply:
Fine (I guess, given https://github.com/php/php-src/blob/7c41dd9c9a96932b7acbd811bfb7d1cca7c142cc/ext/random/randomizer.c#L97-L99 and https://github.com/php/php-src/blob/7c41dd9c9a96932b7acbd811bfb7d1cca7c142cc/ext/random/engine_user.c#L36-L40 and that `sizeof(zend_long)` is equal to `sizeof(uint64_t)` on 64-bit systems), but as I said, there were *2* occurrences in that file (compatibility_user.phpt), but you only removed the first one; did you "forget" to also remove the second one (https://github.com/php/php-src/blob/7c41dd9c9a96932b7acbd811bfb7d1cca7c142cc/ext/random/tests/03_randomizer/compatibility_user.phpt#L72-L74) or is it somewhat useful for `Xoshiro256StarStar` (although it apparently wasn't for `PcgOneseq128XslRr64`)?

Reply:
Thank you for your calm and measured point of view. I was totally confused. From what I see, the guard test is unnecessary as it is tested in get_bytes.phpt. I have removed the if completely.

fixed: 5144026